direct/functionals/hfen.py

```diff
@@ -0,0 +1,320 @@
+# coding=utf-8
+# Copyright (c) DIRECT Contributors
+#
+
+from __future__ import annotations
+
+import math
+from typing import Optional
+
+import numpy as np
+import torch
+from torch import nn
+
+__all__ = ["hfen_l1", "hfen_l2", "HFENLoss", "HFENL1Loss", "HFENL2Loss"]
+
+
+def get_log_kernel2d(kernel_size: int | list[int] = 5, sigma: Optional[float | list[float]] = None) -> torch.Tensor:
+    """Generates a 2D LoG (Laplacian of Gaussian) kernel.
+
```

Comment:
If you do not intend to export, I would propose to prepend the modules with a `_`, this way they don't end up in the documentation and python linters will give a warning when you try to import it from another module.